init-tools.cmd

```diff
@@ -71,6 +71,8 @@ if NOT exist "%BUILD_TOOLS_PATH%init-tools.cmd" (
 :afterbuildtoolsrestore
 
 echo Initializing BuildTools...
+:: Setting RunnintInitTools to true since the restore of project.csproj will throw some warnings for package downgrades and we want to disable TreatWarningsAsErrors.
+set RunningInitTools=true
```

Comment:
I'm not crazy about this....

Reply:
Agreed, but the problem is that TreatWarningsAsErrors is set to true for all projects on core-setup by dir.props, and the new CLI will throw warnings for package downgrades closure of the tool-runtime. The other way to do this would be to have a change in buildtools, but I think this would be more scoped.

Reply:
Which packages are causing the downgrades?  Is it only `Cci`?

/cc @weshaggard

Reply:
I'm not crazy about this either and I'm also interested in the downgrade warnings

Reply:
Yup looks like it's mostly the CCI ones plus one saying that one package used Desktop TFM instead of netcoreapp.
```
  Restoring packages for /Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj...
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Collections.NonGeneric (>= 4.0.1-rc4-24217-00) but System.Collections.NonGeneric 4.0.1-rc4-24217-00 was not found. An approximate best match of System.Collections.NonGeneric 4.0.1-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.IO.FileSystem.Primitives (>= 4.0.1-rc4-24217-00) but System.IO.FileSystem.Primitives 4.0.1-rc4-24217-00 was not found. An approximate best match of System.IO.FileSystem.Primitives 4.0.1-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Collections (>= 4.0.11-rc4-24217-00) but System.Collections 4.0.11-rc4-24217-00 was not found. An approximate best match of System.Collections 4.0.11-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Diagnostics.Debug (>= 4.0.11-rc4-24217-00) but System.Diagnostics.Debug 4.0.11-rc4-24217-00 was not found. An approximate best match of System.Diagnostics.Debug 4.0.11-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Globalization (>= 4.0.11-rc4-24217-00) but System.Globalization 4.0.11-rc4-24217-00 was not found. An approximate best match of System.Globalization 4.0.11-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.IO (>= 4.1.0-rc4-24217-00) but System.IO 4.1.0-rc4-24217-00 was not found. An approximate best match of System.IO 4.1.0-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.IO.FileSystem (>= 4.0.1-rc4-24217-00) but System.IO.FileSystem 4.0.1-rc4-24217-00 was not found. An approximate best match of System.IO.FileSystem 4.0.1-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Reflection (>= 4.1.0-rc4-24217-00) but System.Reflection 4.1.0-rc4-24217-00 was not found. An approximate best match of System.Reflection 4.1.0-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Resources.ResourceManager (>= 4.0.1-rc4-24217-00) but System.Resources.ResourceManager 4.0.1-rc4-24217-00 was not found. An approximate best match of System.Resources.ResourceManager 4.0.1-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Runtime (>= 4.1.0-rc4-24217-00) but System.Runtime 4.1.0-rc4-24217-00 was not found. An approximate best match of System.Runtime 4.1.0-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Runtime.Extensions (>= 4.1.0-rc4-24217-00) but System.Runtime.Extensions 4.1.0-rc4-24217-00 was not found. An approximate best match of System.Runtime.Extensions 4.1.0-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Runtime.InteropServices (>= 4.1.0-rc4-24217-00) but System.Runtime.InteropServices 4.1.0-rc4-24217-00 was not found. An approximate best match of System.Runtime.InteropServices 4.1.0-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Text.Encoding (>= 4.0.11-rc4-24217-00) but System.Text.Encoding 4.0.11-rc4-24217-00 was not found. An approximate best match of System.Text.Encoding 4.0.11-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1603: Microsoft.Cci 4.0.0-rc4-24217-00 depends on System.Threading (>= 4.0.11-rc4-24217-00) but System.Threading 4.0.11-rc4-24217-00 was not found. An approximate best match of System.Threading 4.0.11-rc4-24217-01 was resolved.
/Users/joperezr/git/core-setup/packages/microsoft.dotnet.buildtools/2.0.0-servicing-02104-04/lib/tool-runtime/project.csproj : error NU1701: Package 'Microsoft.Net.Compilers.Targets.NetCore 0.1.5-dev' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
```

Reply:
I'm fine if you guys rather have me fix the package dependencies instead if that is what we want.

Reply:
I think the cleanest thing to do here would be to fix the dependencies.  But I don't think that is feasible at this moment.

The next step down would be to suppress these warnings in project.csproj in buildtools.  Outside repos shouldn't have to do workarounds for buildtools' problems.  buildtools should fix them itself.

Reply:
sounds good, I'll fix that from buildtools side then

Reply:
PR for buildtools changes are up: https://github.com/dotnet/buildtools/pull/1731  

Once that is merged and buildtools is available I'll update this PR.

Reply:
While I'm fine with disabling this in the project directly. For the CCI package warnings we could also fix it by uploading those packages to nuget.org. Ultimately we want to update the CCI package but we could fix it by uploading those dependent packages.